debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -991,11 +995,10 @@ public void readSchema(Tables tables, String databaseCatalog, String schemaNameP
         }
     }
 
-    protected List<ColumnEditor> readTableColumns(ResultSet rs, TableId tableId, ColumnNameFilter columnFilter) throws SQLException {
-        final List<ColumnEditor> cols = new ArrayList<>();
+    protected Optional<ColumnEditor> readTableColumn(ResultSet rs, TableId tableId, ColumnNameFilter columnFilter) throws SQLException {
```

Comment:
Why does it return just a single column now? That's surprising given that the filter might accept more than one column?

Reply:
@gunnarmorling If you look at the caller then you'll see that's the one who is responsible for JDBC column list iteration. So this method process only one record in result set that is passed to it

Reply:
https://github.com/debezium/debezium-incubator/pull/33/files#diff-ebdcb9cc29d754d5aeac500272172556R361
https://github.com/debezium/debezium/pull/651/files#diff-6d089af7912259a716a8266f7cac7835R963
https://github.com/debezium/debezium-incubator/pull/33/files#diff-ebdcb9cc29d754d5aeac500272172556R381

Reply:
Ok, got it now. I was put off by `ResultSet` being used as parameter, which doesn't really make clear what's in it, and the parameter name doesn't help either. I'll polish a bit and merge.